File: CS/WinForms_Dashboard_Drill-Down/CustomPropertyExtensions.cs
using DevExpress.DashboardCommon;
using System;

namespace WinForms_Dashboard_Drill_Down {
    public static class CustomPropertyExtensions {
        public static T GetValue<T>(this CustomProperties property, string name) where T : struct {
            var value = property.GetValue(name);
            if(value == null) return default(T);
            return (T)Convert.ChangeType(value, typeof(T));
        }
    }
}

File: VB/WinForms_Dashboard_Drill-Down/CustomPropertyExtensions.vb
Imports DevExpress.DashboardCommon
Imports System

Namespace WinForms_Dashboard_Drill_Down
	Public Module CustomPropertyExtensions
		<System.Runtime.CompilerServices.Extension> _
		Public Function GetValue(Of T As Structure)(ByVal [property] As CustomProperties, ByVal name As String) As T
			Dim value = [property].GetValue(name)
			If value Is Nothing Then
				Return CType(Nothing, T)
			End If
			Return DirectCast(Convert.ChangeType(value, GetType(T)), T)
		End Function
	End Module
End Namespace

